pkg/streamingpromql/functions/common.go

```diff
@@ -43,3 +46,21 @@ func FloatTransformationDropHistogramsFunc(transform func(f float64) float64) In
 func Passthrough(seriesData types.InstantVectorSeriesData, _ *pooling.LimitingPool) (types.InstantVectorSeriesData, error) {
 	return seriesData, nil
 }
+
+// RangeVectorStepFunction is a function that operates on a range vector step.
+//
+// floatBuffer and histogramBuffer will contain the values for the step range.
+// They may also contain values outside of the step, so step.RangeEnd needs to be considered.
+//
+// Parameters:
+//   - step: the range vector step data to be processed.
+//   - rangeSeconds: the duration of the range in seconds.
```

Comment:
[nit] Not something we have to do in this PR, but I wonder if it makes sense to include `rangeSeconds` in `RangeVectorStepData` and drop the parameter from this function (or just expect the few functions that need this to compute it themselves from `RangeVectorStepData.RangeEnd - RangeVectorStepData.RangeStart` perhaps?)